python/iceberg/api/expressions/expression_parser.py

```diff
@@ -44,7 +44,7 @@
 ident = Word(alphas, alphanums + "_$").setName("identifier")
 columnName = delimitedList(ident, ".", combine=True).setName("column name")
 
-binop = oneOf("= == != < > >= <= eq ne lt le gt ge <>", caseless=False)
+binop = oneOf("= == != < > >= <= eq ne lt le gt ge <> startsWith", caseless=False)
```

Comment:
Would it make sense to allow for `starts_with` in the expression parser too? The SQL wouldn't exactly be portable outside of PySpark etc, but users might want to use the language natural `starts_with` and we already have `lt` `le` and their counterparts `<` `<=` etc.

Reply:
> I think there might be some arrow stuff to add, but that might be outdated. A few comments.

Was there something you are thinking of specifically? I think just the dataset filter translation  is all that's needed for what's currently in the read path.  I could be missing something though.

Reply:
Sure that sounds like a reasonable suggestion